e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+    ) -> None:
+        expect_attr(self.loc, "data-grid", value=value, timeout=timeout)
+
+    def expect_sep_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-prettify-separator", value=value, timeout=timeout)
+
+    def expect_pre_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-prefix", value=value, timeout=timeout)
+
+    def expect_post_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-postfix", value=value, timeout=timeout)
+
+    # def expect_data_type_to_have_value(
+    #     self, value: AttrValue, *, timeout: Timeout = None
+    # ) -> None:
+    #     expect_attr(self.loc, "data-data-type", value=value, timeout=timeout)
+
+    def expect_time_format_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-time-format", value=value, timeout=timeout)
+
+    def expect_timezone_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-timezone", value=value, timeout=timeout)
+
+    def expect_drag_range_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc, "data-drag-interval", value=value, timeout=timeout)
+
+
+def _date_str(date: typing.Union[datetime.date, AttrValue]) -> OptionalStr:
+    if date is None:
+        return None
+    elif isinstance(date, typing.Pattern):
+        return date.pattern
+    elif isinstance(date, datetime.date):
+        return str(date)
+    else:
+        return str(datetime.date.fromisoformat(date))
+
+
+class _DateBase(_WidthContainer, _InputWithLabel):
+    # id: str,
+    # label: TagChildArg,
+    # value: Optional[Union[date, str]] = None,
+    # min: Optional[Union[date, str]] = None,
+    # max: Optional[Union[date, str]] = None,
+    # format: str = "yyyy-mm-dd",
+    # startview: str = "month",
+    # weekstart: int = 0,
+    # language: str = "en",
+    # width: Optional[str] = None,
+    # autoclose: bool = True,
+    # datesdisabled: Optional[List[str]] = None,
+    # daysofweekdisabled: Optional[List[int]] = None,
+    def set(
+        self: _InputWithContainerP,
+        value: typing.Union[datetime.date, str, None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        value_str = _date_str(value)
+        if value_str is None:
+            self.loc.fill("", timeout=timeout)
+            return
+
+        self.loc.fill(value_str, timeout=timeout)
+        # TODO-barret; How to trigger the update without opening the date picker?
+        self.loc.evaluate('(el) => $(el).bsDatepicker("update");')
+
+    def expect_value(
+        self,
+        value: typing.Union[datetime.date, AttrValue],
```

Comment:
Due to `language` parameter of `dateInput`, don't allow `datetime.date`, instead have the caller provide the exact string

Reply:
Same for the rest of the class